plugins/inputs/t128_tank/reader.go

```diff
@@ -221,7 +225,9 @@ func (r *Reader) read(readCtx context.Context, startingIndex index) {
 		time.Sleep(r.restartDelay)
 	}
 
-	r.readDone <- err
+	defer func() {
```

Comment:
revert this change. The function is currently exiting, so there is no reason to defer an operation...